atoma-service/src/handlers/embeddings.rs

```diff
@@ -116,6 +117,115 @@ pub async fn embeddings_handler(
     }
 }
 
+/// OpenAPI documentation structure for the embeddings endpoint.
+///
+/// This struct defines the OpenAPI (Swagger) documentation for the embeddings API,
+/// including all request and response schemas. It uses the `utoipa` framework to generate
+/// the API documentation.
+#[derive(OpenApi)]
+#[openapi(paths(confidential_embeddings_handler))]
+pub(crate) struct ConfidentialEmbeddingsOpenApi;
+
+/// Handler for confidential embeddings requests
+///
+/// This endpoint processes embedding requests with additional confidential computing guarantees.
+/// It forwards the request to the embeddings service and returns an encrypted response that can
+/// only be decrypted by the client.
+///
+/// # Arguments
+///
+/// * `request_metadata` - Extension containing request context like stack ID and encryption details
+/// * `state` - Application state containing service URLs and shared resources
+/// * `payload` - The embedding request body as JSON
+///
+/// # Returns
+///
+/// Returns a `Result` containing either:
+/// * `Json<Value>` - The encrypted embeddings response
+/// * `AtomaServiceError` - Error details if the request processing fails
+///
+/// # Errors
+///
+/// Returns `AtomaServiceError::InternalError` if:
+/// * The embeddings service request fails
+/// * Response processing or encryption fails
+/// * Stack compute unit updates fail
+///
+/// # Example Request
+///
+/// ```json
+/// {
+///     "model": "text-embedding-ada-002",
+///     "input": "The quick brown fox jumps over the lazy dog"
+/// }
+/// ```
+#[utoipa::path(
+    post,
+    path = "",
+    tag = "confidential-embeddings",
+    request_body = ConfidentialComputeRequest,
+    responses(
+        (status = OK, description = "Confidential embeddings generated successfully", body = ConfidentialComputeResponse),
+        (status = INTERNAL_SERVER_ERROR, description = "Internal server error")
+    )
+)]
+#[instrument(
+    level = "info",
+    skip(state, payload),
+    fields(path = request_metadata.endpoint_path)
+)]
+pub async fn confidential_embeddings_handler(
+    Extension(request_metadata): Extension<RequestMetadata>,
+    State(state): State<AppState>,
+    Json(payload): Json<Value>,
+) -> Result<Json<Value>, AtomaServiceError> {
+    info!("Received embeddings request, with payload: {payload}");
+    let model = payload
+        .get("model")
```

Comment:
in the other file you introduced `STREAM_KEY`. Maybe it would be nice to have `MODEL_KEY` as well.

Reply:
Makes sense